packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .bundle.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_ca_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_ca.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert_key.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert_key.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/secrets.env"
+        env         = true
+        change_mode = "restart"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+TFE_LICENSE                             = {{ .tfe_license }}
+TFE_DATABASE_PASSWORD                   = {{ .db_password }}
+TFE_OBJECT_STORAGE_S3_SECRET_ACCESS_KEY = {{ .s3_secret_key }}
+TFE_REDIS_PASSWORD                      = {{ .redis_password }}
+TFE_ENCRYPTION_PASSWORD                 = {{ .tfe_encryption_password }}
+TFE_IMAGE_REGISTRY_PASSWORD             = {{ .tfe_image_registry_password }}
+{{- end -}}
+EOF
+      }
+
+      config {
+         image = [[ .tfe_fdo_nomad.tfe_image | quote ]]
+         auth {
+            # User Input is mandatory.
+            username = [[ .tfe_fdo_nomad.tfe_image_registry_username | quote ]]
+            # User Input is mandatory.
+            password       = "${TFE_IMAGE_REGISTRY_PASSWORD}"
+            server_address = [[ .tfe_fdo_nomad.tfe_image_server_address | quote ]]
+         }
+         ports = ["tfe", "http", "vault"]
+
+         volumes = [
+          "secrets:[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]",
+        ]
+  
+      }
+
+      env {
+        
+        TFE_RUN_PIPELINE_DRIVER = "nomad"
+        # User Input is mandatory. 
+        TFE_RUN_PIPELINE_NOMAD_ADDRESS             = [[ .tfe_fdo_nomad.tfe_run_pipeline_nomad_address | quote ]]
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_INSECURE = [[ .tfe_fdo_nomad.tfe_run_pipeline_nomad_tls_config_insecure | quote ]]
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CA_CERT = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_ca_cert.pem"
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CLIENT_CERT = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_cert.pem"
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CLIENT_KEY = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_cert_key.pem"
```

Comment:
These certs aren't in the path we mounted from the host, they're being brought in as template secrets to the `/secrets` dir, aren't they?

Reply:
Yes correct, we get the certs from Nomad variables, then copy them in `secrets` dir and mount it to TFE container.

Reply:
Yes, but `[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]` is not the Nomad `secrets` dir. So you're setting this environment variable to a file that doesn't exist.

Reply:
Yes, in `/secrets`, as you've shown right here. Not in `"[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]"`, which is where you're telling TFE to expect them.

Reply:
we mount it in the job file

`   volumes = [
          "secrets:[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]",
        ]
  `

Reply:
Oh, wait... what is this?

```
         volumes = [
          "secrets:[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]",
        ]
```

Are you rebinding the secrets tmpfs somewhere else? Why would we need to do that instead of just using `/secrets`? And if that's the only volume we're mounting, how are we persisting data?

Reply:
The `/secrets` folder is a bind-mount on the container instance. See [File System concepts](https://developer.hashicorp.com/nomad/docs/concepts/filesystem). You can see this for yourself by looking at the container via `docker inspect`. The allocation logs will tell you what the problem is, presumably.